feat: Configurable backpack open

Chappie74 opened the pull request:
**Description**

Added functionality for configuring whether the backpack should open if empty. 

**Testing**

When instantiating the backpack, add  `allowEmptyBackpackOpen: false` as an option. This defaults to true, which keeps the default blockly behaviour.

maribethb commented on plugins/workspace-backpack/src/options.js:
```diff
@@ -44,6 +45,7 @@ export function parseOptions(options) {
       copyAllToBackpack: false,
       pasteAllToBackpack: false,
       disablePreconditionChecks: false,
+      allowEmptyBackpackOpen: true,
```
This should be a new top-level option instead of being nested under contextMenu options. The context menu options configures which actions are available in the context menu, and this option is unrelated.

maribethb commented on plugins/nominal-connection-checker/src/type_hierarchy.js:
```diff
@@ -497,16 +497,17 @@ export class TypeHierarchy {
-   * :!Array<!TypeStructure>}
-   *     getParamsForCommon A function that takes in the definition of a type
-   *     and maps the given actual params to the parameters of the common type.
+:!Array<!TypeStructure>}
```
Could you revert all the changes that aren't in the backpack plugin? They aren't related and some of the comments are incorrectly formatted - e.g. not starting the comment line with the asterisk can cause problems with jsdoc parsing.

Chappie74 replied:
Yeah I was wondering about all those changes. I couldn't get the project to run due to linting errors. And I ran the lint:fix command, which changed a lot of files.

maribethb commented on plugins/workspace-backpack/src/backpack.js:
```diff
@@ -578,7 +578,8 @@ export class Backpack extends Blockly.DragTarget {
   isOpenable_() {
-    return !this.isOpen();
+    return !this.isOpen() &&
+    this.options_.contextMenu.allowEmptyBackpackOpen? true: this.getCount() > 0;
```
Surprised the linter didn't complain about this, but could you add a space before the `?` and `:` please: https://google.github.io/styleguide/jsguide.html#:~:text=On%20both%20sides%20of%20any%20binary%20or%20ternary%20operator.

and then this line will probably need to be wrapped to stay under 80 chars

maribethb reviewed (commented):
Thank you for the PR and for contributing to Blockly! I think this just needs a bit of cleanup and then we should be good to go. Let me know if you have any questions!

Could you also add information about the new option in the README for this plugin?

Chappie74 committed:
- Added functionality for configuring whether the backpack should open if empty.

Chappie74 committed:
feat: Added functionality for configuring whether the backpack should open if empty.

When instantiating the backpack, add allowEmptyBackpackOpen: false as an option. This defaults to true, which keeps the default blockly behaviour.

Chappie74 committed:
Merge remote-tracking branch 'origin/blockly-backpack-openable' into blockly-backpack-openable
